### ISSUE TITLE
plateau: [dev branch] various issues with search

### ISSUE THREAD

**Issue by ReesePlews** [metanorma/firelight#23]
search is working much better in the dev branch than earlier, but there are still some questions about search rules; the anticipated results are not always what one expects.

input of "c" returns 
![image](https://github.com/user-attachments/assets/beb8e8e4-9169-466f-bc6e-409c10a5e540)

but input of "park"  or "_park" returns
![image](https://github.com/user-attachments/assets/179a9dcf-aae6-45d5-9a67-55ba5808f2ac)

"*park" returns
![image](https://github.com/user-attachments/assets/f2a4dc15-c35d-4289-b613-7a45b5d63e83)

how to pick up "park" to get better results?

**Comment by strogonoff** [comment]
Good point. It looks like tokenizer should be improved to split camelCased words into distinct tokens (both in index and query). cc @ronaldtse this is a search problem that could probably be delegated.

**Comment by strogonoff** [comment]
> input of "c" returns

I am not sure if that example shows a problem? @ReesePlews 

Regarding the park, that is a problem. I will rename the issue to reflect that problem.

**Comment by strogonoff** [comment]
@ReesePlews The reason “c” is found in clause “4.25.4.6.9”, for example, is because you will see “C” does appear as a standalone word in the table.
